### DISCUSSION TITLE
Slow loading when opening the dashboard.

### DISCUSSION THREAD

**Discussion by WarC0zes** [Clooos/Bubble-Card#293] [Feature requests]
Hi,
I noticed the dashboard loading slowly when I access it. I see all the popups cards appear on the screen and disappear.
Same behavior on PC or mobile.
![bubble4](https://github.com/Clooos/Bubble-Card/assets/91471397/f414bb78-74d8-4a87-bcd6-204b06704cb1)

I'm using v1.2.0.
Need additional information?

**Comment by Clooos** [comment]
Hi and thanks for your feedback! This behavior is expected on an empty dashboard because the cards are loaded on your page before being converted to pop-ups, a possible workaround is to have some cards on your dashboard that are not into a pop-up and that are positioned before your pop-ups, or an other workaround is to have full Bubble Card pop-ups, this is only happening with other cards. But I'm trying to find a solution to that behavior since the first beta.

And what do you mean by loading slowly, it was faster before v1.2.0? 

Edit: A new version will come really soon that is faster than 1.2.0.

**Comment by WarC0zes** [comment]
I said slow loading, but that was to say that I'm seeing maps loading that I didn't have before.
I will try your solution.

**Comment by Clooos** [comment]
I've just released a new version, can you tell me if you see an improvement?

**Comment by WarC0zes** [comment]
Hi,
with version 1.2.3, it's good on mobile but still the same on PC.
Maybe my setup with 10 horizontal-buttons and 10 popups are too heavy?

my config pc:
win10
i7 9700
16Go DDR4
Chrome navigator (last version).

**Comment by Clooos** [comment]
Have you tried to clear your PC browser cache? Because your config should be way faster than your mobile.

**Comment by WarC0zes** [comment]
> Have you tried to clear your PC browser cache? Because your config should be way faster than your mobile.

Yes, cache clear.
![bubble6](https://github.com/Clooos/Bubble-Card/assets/91471397/13f1f583-f058-4f7f-8b3e-caeecb3801e0)

on mobile, I see 1 column loading, then the screen becomes empty with the clock.
on pc, I see all the columns loading, then the screen goes blank with the clock.
between version 1.2.0 and 1.2.3, performance level.

**Comment by Clooos** [comment]
Can you tell me what was the last version that was working for you without this behavior? And can you try to reinstall it to see if it works like it should?

**Comment by WarC0zes** [comment]
1.1.2 is OK.
![bubble7](https://github.com/Clooos/Bubble-Card/assets/91471397/79f71728-0802-46f8-a808-048d4d134619)
from 1.2.0,it's slow and you can see the columns loading.

**Comment by WarC0zes** [comment]
Hi,
1.2.4 same problem.
I tried with chrome and edge, I cleared the cache. I don't see any errors in the Chrome console.

**Comment by Clooos** [comment]
Can you tell me if it's working in v1.3.0?

**Comment by WarC0zes** [comment]
no, same as the version 1.2.4.

**Comment by WarC0zes** [comment]
same as the version 1.3.3 on desktop with chrome and edge.
![bubble9](https://github.com/Clooos/Bubble-Card/assets/91471397/fb393397-a428-4a62-8635-d54f1a886112)

**Comment by WarC0zes** [comment]
Hi,
same as the version 1.4.2 on desktop with chrome and edge.
On mobile, it's better but I can see columns loading sometimes.

**Comment by Clooos** [comment]
Hi! Here is a new beta version, can you please try to follow the instructions to configure it? This should totally fix your issue 🙂 

https://github.com/Clooos/Bubble-Card/releases/edit/v1.5.0-beta.1

**Comment by WarC0zes** [comment]
Hi,
Sorry but it's still the same. I followed your indication for the beta, clear the cache...
![bubble13](https://github.com/Clooos/Bubble-Card/assets/91471397/dbb7c6ed-d755-43f3-9932-263978e0c1d1)

On my mobile (android 13) with the HA application it's faster than on the PC with Chrome, but I see a page load too.

**Comment by Clooos** [comment]
I see that you have an error with card-mod, I guess that you've added it as a frontend module too (I had this error as well). Just remove card-mod from resources and you should not have this error anymore, this is possible that this is why you have still this issue 🤞

**Comment by WarC0zes** [comment]
Unfortunately it's not a card-mod problem.
![bubble14](https://github.com/Clooos/Bubble-Card/assets/91471397/a046cf93-3ef1-4d7f-a09e-4bf4832f5f14)

I have an error message in HACS, as I deleted the resources. Can I ignore it?
![bubble error hacs](https://github.com/Clooos/Bubble-Card/assets/91471397/21dda962-5a6a-4a04-9bb9-13e0f80f9560)

**Comment by Clooos** [comment]
Yes I have this too, and I also have the card-mod error (but only once) like you now, are you sure that you don't have any other error in your console? It's possible that this is a conflict with one of your cards, but this issue should be gone in this beta so I really don't understand why you are still experiencing this right now 😅

**Comment by WarC0zes** [comment]
No, the 2 errors you see, its from apexcharts-card.
Error: <g> attribute transform: Expected number, "translate(NaN, 0)".

**Comment by Clooos** [comment]
This should be THE one 🤞
https://github.com/Clooos/Bubble-Card/releases/tag/v1.5.0-beta.2

**Comment by WarC0zes** [comment]
Hi,
![bubble15](https://github.com/Clooos/Bubble-Card/assets/91471397/a361b30f-b62f-43ca-934d-b252b343347c)
There is an error in the chrome console.
![bubble error console beta2](https://github.com/Clooos/Bubble-Card/assets/91471397/009f529d-633d-41bd-9923-e32d7ad707f3)


I may have misunderstood the readme, but the bubble-pop-up card is not added to the bubble-card folder. I downloaded the source code and manually copied the bubble-pop-up.js file.

**Comment by Clooos** [comment]
Bubble Card should not be added as an extra module in this version, only bubble-pop-up.js, this is why you have this error. And indeed the file is not included in the package (my bad).

**Comment by WarC0zes** [comment]
Ok, I correct my config and test

**Comment by WarC0zes** [comment]
Config OK, no errors in console.
But that doesn't change anything, it's still slow.
I cleared the cache of chrome.

**Comment by WarC0zes** [comment]
Would another card be the problem?
![card use](https://github.com/Clooos/Bubble-Card/assets/91471397/27b05593-0521-467f-8cca-fd0451619d83)
![card use2](https://github.com/Clooos/Bubble-Card/assets/91471397/59c08a90-4578-428b-b519-20bf860da006)

**Comment by Clooos** [comment]
I have only 7 custom cards in my setup, and you have a lot of them, it's most likely because they all have to be loaded before the pop-ups initialization. I still don't understand why it's working for you in 1.0.3.

But I will soon try to add something to hide the dashboard during the pop-ups initialization, I will tell you when it will be available! 🤞

**Comment by WarC0zes** [comment]
no change with 1.5.3, shame.

**Comment by Clooos** [comment]
Have you tried with a new dashboard to see what custom card is causing this? Because your issue looks like some kind of conflict with a custom card.

And also are you in Optimized mode?

**Comment by WarC0zes** [comment]
> Have you tried with a new dashboard to see what custom card is causing this? Because your issue looks like some kind of conflict with a custom card.

No, I didn't try. I will try.



> And also are you in Optimized mode?

Yes.
I have the popup card, in the card selection.
